src/main/java/org/example/jvspringbootfirstbook/service/BookServiceImpl.java

```diff
@@ -48,4 +52,11 @@ public BookDto update(Long id, CreateBookRequestDto createBookRequestDto) {
         bookRepository.updateBookById(id, model);
         return bookMapping.toBookDto(model);
     }
+
+    @Override
+    public List<BookDto> searchBooks(BookSearchParametersDto searchParameters) {
+        Specification<Book> bookSpecification = bookSpecificationBuilder.build(searchParameters);
+        List<Book> bookList = bookRepository.findAll(bookSpecification);
+        return bookList.stream().map(bookMapping::toBookDto).toList();
```

Comment:
There is no need to create this temporary variable. Please inline these two lines, For example like this:

```suggestion
        return bookRepository.findAll(bookSpecification).stream()
                .map(bookMapper::toDto)
                .toList();
```